toolchain/check/merge.cpp

```diff
@@ -12,6 +12,135 @@
 
 namespace Carbon::Check {
 
+CARBON_DIAGNOSTIC(RedeclPrevDecl, Note, "Previously declared here.");
+
+// Diagnoses a redundant redeclaration diagnostic.
+static auto DiagnoseRedeclRedundant(Context& context, Lex::TokenKind decl_kind,
```

Comment:
Optional: the original name `DiagnoseRedundantRedecl` reads more naturally to me (notice how it corresponds more directly to the comment, for example).

Reply:
Dropped "redecl", since all of these are redeclaration diagnostics. In this case, I'm putting more weight on keeping the names of diagnostics consistent cross-diagnostic (trying to highlight the cluster of redeclaration diagnostics), and the function names consistent with diagnostics.